lib/bill_forward/entities/invoice.rb

```diff
@@ -35,6 +35,44 @@ def get_by_account_id(id, query_params = {}, custom_client = nil)
 
         self.request_many('get', endpoint, query_params, custom_client)
       end
+
+      def recalculate(id, query_object = {}, custom_client = nil)
+        raise ArgumentError.new("id cannot be nil") if id.nil?
+
+        endpoint = sprintf('%s/recalculate',
+                           ERB::Util.url_encode(id)
+        )
+
+        request_entity = BillForward::GenericEntity.new(
+            query_object
+        )
+
+        self.request_first('post', endpoint, request_entity, nil, custom_client)
+      end
+
+      def execute(id, query_object = {}, custom_client = nil)
+        raise ArgumentError.new("id cannot be nil") if id.nil?
+
+        endpoint = sprintf('%s/execute',
+                           ERB::Util.url_encode(id)
+        )
+
+        request_entity = BillForward::GenericEntity.new(
+            query_object
+        )
+
+        self.request_first('post', endpoint, request_entity, nil, custom_client)
+      end
+
+      def get_charges(id, query_params = {}, custom_client = nil)
+        raise ArgumentError.new("id cannot be nil") if id.nil?
+
+        endpoint = sprintf('%s/charges',
+                           ERB::Util.url_encode(id)
+        )
+
+        self.request_many('get', endpoint, query_params, custom_client)
```

Comment:
This will return a list of Invoice. you should return a list of SubscriptionCharge.